controlpanel/api/models/tool.py

```diff
@@ -27,6 +27,10 @@ class Tool(TimeStampedModel):
     name = models.CharField(max_length=100, blank=False)
     values = JSONField(default=dict)
     version = models.CharField(max_length=100, blank=False)
+    # The release is restricted to only certain users.
+    is_restricted = models.BooleanField(default=False)
```

Comment:
I haven't got to the meat of the PR yet (the logic to show/hide tools versions to the users) so maybe this field is here to simplify that logic/query but isn't a release restricted when there are 1+ target_users associated with it?

What I'm wondering is if this field is redundant (just writing it down, maybe it's a silly observation will clear itself once I get to the logic using it)

Reply:
Good question and I thought long and hard about it. In the end I went with:

* This is explicit.
* This is unambiguous.
* The name of the flag says what it means (thus helping future self/maintainers).

I also think it makes the logic simpler to understand when used elsewhere - you're checking a boolean flag rather than comparing values.

In any case, this column is a negligible addition but has the benefits described above. Hope this makes sense..! :+1:

Reply:
Yeah I saw the logic and I think the logic with `Q` and this flag is probably easier than having to select records with no associated stuff.

Cool, happy with this field 👍